File: app/src/main/java/com/an/github/data/local/AppDatabase.java
package com.an.github.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import com.an.github.data.local.converter.TimestampConverter;
import com.an.github.data.local.dao.GithubDao;
import com.an.github.data.local.entity.GithubEntity;


@Database(entities = {GithubEntity.class}, version = 1,  exportSchema = false)
@TypeConverters({TimestampConverter.class})
public abstract class AppDatabase extends RoomDatabase {

    public abstract GithubDao githubDao();
}

File: app/src/main/java/com/an/github/data/local/AppDatabase.kt
package com.an.github.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import com.an.github.data.local.dao.GithubDao
import com.an.github.data.local.dao.GithubRemoteKeyDao
import com.an.github.data.local.entity.GithubEntity
import com.an.github.data.local.entity.GithubRemoteKey

@Database(
    entities = [GithubEntity::class, GithubRemoteKey::class],
    version = 1,
    exportSchema = false
)
abstract class AppDatabase: RoomDatabase() {
    abstract val githubDao: GithubDao
    abstract val githubKeyDao: GithubRemoteKeyDao
}
